src/views/branchNode.ts

```diff
@@ -24,6 +25,19 @@ export class BranchNode extends ExplorerRefNode {
         return this.branch.name;
     }
 
+    get label(): string {
+        const branchName = this.branch.getName();
+        if (this.explorer.config.branches.layout === ExplorerBranchesLayout.List) {
+            return branchName;
+        } else {
+            return !!branchName.match(/\s/) ? branchName : this.branch.getBasename();
```

Comment:
I'm not sure I understand the logic here. Why are you checking if the name contains any whitespace?

Reply:
In some case like detached git state, the folder name will not be parsed correctly. So here I see any white space as invalid folder branch name, and choose not to handle them. Same logic appears in `branchesNode.ts:38` and `remoteNode.ts:39`.

Reply:
Ah ok, but why is it needed? A "branch" with spaces could be handled just the same as any other without '/' characters right? Was there an explicit case you were protecting against?

Reply:
Ah -- ok, thanks